src/ios.rs

```diff
@@ -27,14 +26,16 @@ extern "C" {
 pub fn getrandom_inner(dest: &mut [u8]) -> Result<(), Error> {
     let ret = unsafe { SecRandomCopyBytes(kSecRandomDefault, dest.len(), dest.as_mut_ptr()) };
     if ret == -1 {
-        error!("SecRandomCopyBytes call failed");
-        Err(Error::UNKNOWN)
+        Err(Error::from_custom_error(CALL_FAILED))
```

Comment:
I don't know if this case is ever likely to be encountered, but if it is, the new code reports only `UNAVAILABLE` (unless `CALL_FAILED` is supposed to be distinct).

I think we should keep the `error` log? These macros are optimised out completely if logging is not enabled.

Reply:
Now that the codes have been moved to `error.rs`, the `UNAVAILABLE` and `UNKNOWN` errors have been removed. 

As for the `error!` logging. My strategy was to:

- remove logs that just had the same error message as the returned error (to prevent log spew)
- keep logs that added additional context (to aid in debugging)

Do you want me the add the `error!("SecRandomCopyBytes call failed");` lines back?  We're returning an `Error` who's display implementation is `"SecRandomCopyBytes: call failed"`.

Reply:
The motivation for redundant messages is https://github.com/rust-random/rand/issues/837 — i.e. they may not be redundant. But of course we're working on solving that...